go-gen-checksum.sh

```diff
@@ -41,21 +40,34 @@ for GOMODULE in ${GOMODULE_PATHS[@]}; do
     GOMOD_SUM_ENTRY="$GOMOD_NAME $GOMOD_VERSION $GOMOD_SUM"
     GOMOD_DOTMOD_SUM_ENTRY="$GOMOD_NAME $GOMOD_VERSION/go.mod $GOMOD_DOTMOD_SUM"
     echo "----"
-    echo $GOMOD_SUM_ENTRY
-    echo $GOMOD_DOTMOD_SUM_ENTRY
+    echo GOSUM: $GOMOD_SUM_ENTRY
+    echo GOSUM: $GOMOD_DOTMOD_SUM_ENTRY
     echo "----"
     popd
     rm $ROOT_DIR/$GOMOD_PATH/LICENSE
     
     pushd $ROOT_DIR/$GOMODULE
-    go mod edit -require $GOMOD_NAME@$GOMOD_VERSION
-    # mv go.sum go.sum.old
-    # grep -v "$GOMOD_NAME $GOMOD_VERSION" go.sum.old > go.sum
-    echo $GOMOD_SUM_ENTRY >> go.sum
-    echo $GOMOD_DOTMOD_SUM_ENTRY >> go.sum
+    UPDATE=false
+    (cat go.mod | grep -q "$GOMOD_NAME $GOMOD_VERSION") || UPDATE=True
+    if $UPDATE; then
+      go mod edit -require $GOMOD_NAME@$GOMOD_VERSION
+    else
+      echo "ERROR: Version $GOMOD_VERSION already there in go.mod, skipping $GOMOD_PATH in $GOMODULE"
+    fi
+    UPDATE=false
+    (cat go.sum | grep -q "$GOMOD_SUM_ENTRY") || UPDATE=True
+    (cat go.sum | grep -q "$GOMOD_DOTMOD_SUM_ENTRY") || UPDATE=True
+    if $UPDATE; then
+      # mv go.sum go.sum.old
+      # grep -v "$GOMOD_NAME $GOMOD_VERSION" go.sum.old > go.sum
+      echo $GOMOD_SUM_ENTRY >> go.sum
+      echo $GOMOD_DOTMOD_SUM_ENTRY >> go.sum
+    else
+      echo "ERROR: Version $GOMOD_VERSION already there in go.sum, skipping $GOMOD_PATH in $GOMODULE"
```

Comment:
Is this really an error? If not, just remove that word.

Reply:
It's an error because ideally the version being specified is supposed to be a new version that hasn't been released.
And if that version is already present in go.sum, it means some error occurred in the script.